contrib/babelfishpg_tds/src/backend/tds/tdslogin.c

```diff
@@ -1759,29 +1755,39 @@ TdsClientAuthentication(Port *port)
 			break;
 		case uaMD5:
 		case uaPassword:
-			/*
-			 * If pg_hba.conf specifies that the entry should be authenticated using
-			 * password and the request doesn't contain a password, we should
-			 * throw an error.
-			 */
-			if (!loginInfo->password)
+			/* if sspiLen > 0 then GSS auth is already done at this point */
```

Comment:
How about the following change only?
```
if (loginInfo->sspiLen > 0)
{
	Assert(loginInfo->sspi);

	/* Cleanup sspi data. */
	pfree(loginInfo->sspi);
	loginInfo->sspiLen = 0;
	break;
}

/*
 * If pg_hba.conf specifies that the entry should be authenticated using
 * password and the request doesn't contain a password, we should
 * throw an error.
 */
if (!loginInfo->password)
...

```
The assert is valid as loginInfo->sspi is not null if sspiLen>0. (Check FetchLoginRequest)